.github/workflows/maven.yml

```diff
@@ -30,13 +44,15 @@ jobs:
           path: target/surefire-reports
       - name: Upload Coverage
         uses: actions/upload-artifact@v1.0.0
-        if: always()
+        if: matrix.os == 'ubuntu-latest'
         with:
-          name: Coverage Report
+          name: Coverage Report ${{ matrix.os }}
           path: target/jacoco-report
       - name: Convert Jacoco unit test report to Cobertura
+        if: matrix.os == 'ubuntu-latest'
         run: python3 .github/scripts/cover2cover.py target/jacoco-report/jacoco.xml src/main/java > target/jacoco-report/cobertura.xml
       - name: cobertura-report-unit-test
+        if: matrix.os == 'ubuntu-latest'
         uses: shaguptashaikh/cobertura-action@master
         continue-on-error: true
         with:
```

Comment:
You should follow what we do in Nucleus. Using this reporter means that people outside of AWS are not able to run workflows due to the usage of the github.token value.

https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/.github/workflows/maven.yml#L79-L96

https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/.github/workflows/externalPR.yml